datanode/main.go

```diff
@@ -0,0 +1,92 @@
+// datanode/main.go
+package main
+
+import (
+	"log/slog"
+	"net"
+	"net/http"
+	"net/rpc"
+	"os"
+	"strings"
+
+	"github.com/ZhenbangYou/TinyDFS/tiny-dfs/common"
+)
+
+// Command Line Args:
+// Args[1]: Namenode endpoint (IP:port)
+// Args[2]: Datanode endpoint (IP:port)
+// Args[3]: Optional log file path
+func main() {
+	// Check command line arguments
+	if len(os.Args) < 3 || len(os.Args) > 4 {
+		slog.Error("expect 3 or 4 command line arguments", "actual argument count", len(os.Args))
+		os.Exit(1)
+	}
+
+	// Read and validate Namenode endpoint
+	nameNodeEndpoint := os.Args[1]
+	if !common.IsValidEndpoint(nameNodeEndpoint) {
+		slog.Error("invalid namenode endpoint", "endpoint", nameNodeEndpoint)
+		os.Exit(1)
+	}
+
+	// Read and validate Datanode endpoint
+	dataNodeEndpoint := os.Args[2]
+	if !common.IsValidEndpoint(dataNodeEndpoint) {
+		slog.Error("invalid datanode endpoint", "endpoint", dataNodeEndpoint)
+		os.Exit(1)
+	}
+
+	// Set up slog to log into a file or terminal
+	var logHandler slog.Handler
+	var programLevel = new(slog.LevelVar) // Info by default
+
+	if len(os.Args) == 4 {
+		logFilePath := os.Args[3]
+		logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0666)
+		if err != nil {
+			slog.Error("failed to open log file", "error", err)
+			os.Exit(1)
+		}
+		defer logFile.Close()
+		logHandler = slog.NewJSONHandler(logFile, &slog.HandlerOptions{Level: programLevel})
+	} else {
+		logHandler = slog.NewJSONHandler(os.Stderr, &slog.HandlerOptions{Level: programLevel})
+	}
+
+	slog.SetDefault(slog.New(logHandler))
+	programLevel.Set(slog.LevelDebug)
+
+	// Initialize DataNode
+	dataNode := DataNode{
+		nameNodeEndpoint: nameNodeEndpoint,
+		dataNodeEndpoint: dataNodeEndpoint,
+	}
+	slog.Info("Initialized datanode", "nameNodeEndpoint", nameNodeEndpoint, "dataNodeEndpoint", dataNodeEndpoint)
```

Comment:
I would prefer "NameNode Endpoint" to "nameNodeEndpoint", since this is a string, rather than a variable name